api/app/controllers/spree/api/line_items_controller.rb

```diff
@@ -11,16 +11,15 @@ def new
 
       def create
         variant = Spree::Variant.find(params[:line_item][:variant_id])
-        @line_item = @order.contents.add(
-          variant,
-          params[:line_item][:quantity] || 1,
-          options: line_item_params[:options].to_h
-        )
-
-        if @line_item.errors.empty?
+        begin
+          @line_item = @order.contents.add(
+            variant,
+            params[:line_item][:quantity] || 1,
+            options: line_item_params[:options].to_h
+          )
           respond_with(@line_item, status: 201, default_template: :show)
-        else
-          invalid_resource!(@line_item)
+        rescue ActiveRecord::RecordInvalid => error
+          invalid_resource!(error.record)
```

Comment:
I'm wondering ig this could/should be turned into a `rescue_from` in the API::BaseController